web/onos-gui/src/app/onos-topo/devices-list/devices-list.component.ts

```diff
@@ -84,11 +88,25 @@ export class DevicesListComponent implements OnInit, OnDestroy {
         this.selId = undefined;
     }
 
-    onSort(colname: string) {
-        // not yet implemented
+    onSort(colName: string) {
+        if (this.topoDeviceService.sortParams.firstColName === colName) {
+            this.topoDeviceService.switchSortCol(colName.toLowerCase(), this.topoDeviceService.sortParams.firstCriteriaDir);
+        } else {
+            this.topoDeviceService.switchSortCol(colName.toLowerCase(), 1);
+        }
+        this.sortIcon(colName);
+        this.topoDeviceService.deviceList.set('wakeup', new Device());
```

Comment:
there's some trick to avoid having to do this for map (but I can't remember what it is - we can fix some other time)